### PR TITLE
Group Data Provider: Added missing group mapping iterator.

### PR DESCRIPTION
**Pull request by rcasallas-silabs**
#### Problem
Group Data Provider interface is missing an iterator to get all groups associated with the given fabric

#### Change overview
* Added GroupMapping iterator for all the groups in a fabric.
* Added unit tests for the new iterator.

#### Testing
* Unit tests run successfully.

### CODE REVIEW

File: src/credentials/GroupDataProviderImpl.cpp [hunk 1]
```diff
@@ -799,11 +799,98 @@ CHIP_ERROR GroupDataProviderImpl::RemoveAllGroupMappings(chip::FabricIndex fabri
...
+}
+
+size_t GroupDataProviderImpl::AllGroupMappingsIteratorImpl::Count()
+{
+    // chip::EndpointId endpoint_id = mFirstEndpoint;
+    size_t count          = 0;
+    size_t endpoint_index = 0;
+    EndpointData endpoint_data(mFabric, mFirstEndpoint);
+
+    // Loop throug the fabric's endpoints
```
Comment (tcarmelveilleux) [inline src/credentials/GroupDataProviderImpl.cpp:831]:
```suggestion
    // Loop through the fabric's endpoints
```

File: src/credentials/GroupDataProviderImpl.cpp [hunk 2]
```diff
@@ -799,11 +799,98 @@ CHIP_ERROR GroupDataProviderImpl::RemoveAllGroupMappings(chip::FabricIndex fabri
...
+    mFirstEndpoint = fabric.first_endpoint;
+    mFirstGroup    = true;
+    mEndpointCount = fabric.endpoint_count;
+    mEndpoint      = fabric.first_endpoint;
+    mEndpointIndex = 0;
+}
+
+size_t GroupDataProviderImpl::AllGroupMappingsIteratorImpl::Count()
+{
+    // chip::EndpointId endpoint_id = mFirstEndpoint;
```
Comment (tcarmelveilleux) [inline src/credentials/GroupDataProviderImpl.cpp:826]: Any reason this is commented? Seems like it could be removed

**Review by tcarmelveilleux — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/credentials/GroupDataProviderImpl.cpp (hunk 1 above)
Comment (rcasallas-silabs) [reply src/credentials/GroupDataProviderImpl.cpp]: Fixed.

File: src/credentials/GroupDataProviderImpl.cpp (hunk 2 above)
Comment (rcasallas-silabs) [reply src/credentials/GroupDataProviderImpl.cpp]: Copy-and-paste leftover. Removed.

**Review by yunhanw-google — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**